components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsStorage.kt

```diff
@@ -0,0 +1,202 @@
+     */
+    suspend fun deleteAddress(guid: String): Boolean
+
+    /**
+     * Marks the address with the given [guid] as `in-use`.
+     *
+     * @param guid Unique identifier for the desired address.
+     */
+    suspend fun touchAddress(guid: String)
+}
+
+/**
+ * Information about a credit card.
+ *
+ * @property guid The unique identifier for this credit card.
+ * @property billingName The credit card billing name.
+ * @property cardNumber The credit card number.
+ * @property expiryMonth The credit card expiry month.
+ * @property expiryYear The credit card expiry year.
+ * @property cardType The credit card network ID.
+ */
+data class CreditCard(
+    val guid: String,
+    val billingName: String,
+    val cardNumber: String,
+    val expiryMonth: Int,
+    val expiryYear: Int,
+    val cardType: String
+)
+
+/**
+ * Information about a new credit card.
+ *
+ * @property billingName The credit card billing name.
+ * @property cardNumber The credit card number.
+ * @property expiryMonth The credit card expiry month.
+ * @property expiryYear The credit card expiry year.
+ * @property cardType The credit card network ID.
+ */
+data class NewCreditCardFields(
+    val billingName: String,
+    val cardNumber: String,
+    val expiryMonth: Int,
+    val expiryYear: Int,
+    val cardType: String
+)
+
+/**
+ * Information about a address.
+ *
+ * @property guid The unique identifier for this address.
+ * @property givenName First name.
+ * @property additionalName Middle name.
+ * @property familyName Last name.
+ * @property organization Organization.
+ * @property streetAddress Street address.
+ * @property addressLevel3 Sublocality (Suburb) name type.
+ * @property addressLevel2 Locality (City/Town) name type.
+ * @property addressLevel1 Province/State name type.
+ * @property postalCode Postal code.
+ * @property country Country.
+ * @property tel Telephone number.
+ * @property email E-mail address.
+ */
+data class Address(
+    val guid: String,
+    val givenName: String,
+    val additionalName: String,
+    val familyName: String,
+    val organization: String,
+    val streetAddress: String,
+    val addressLevel3: String,
+    val addressLevel2: String,
+    val addressLevel1: String,
+    val postalCode: String,
+    val country: String,
+    val tel: String,
+    val email: String
+)
+
+/**
+ * Information about a new address.
+ *
+ * @property givenName First name.
+ * @property additionalName Middle name.
+ * @property familyName Last name.
+ * @property organization Organization.
+ * @property streetAddress Street address.
+ * @property addressLevel3 Sublocality (Suburb) name type.
+ * @property addressLevel2 Locality (City/Town) name type.
+ * @property addressLevel1 Province/State name type.
+ * @property postalCode Postal code.
+ * @property country Country.
+ * @property tel Telephone number.
+ * @property email E-mail address.
+ */
+data class NewAddressFields(
```

Comment:
Why do we need _New_ address and _New_ credit card fields? I don't see a reason to have two identical data classes

Reply:
It's not the same thing since Address and CreditCard has an assigned guid when it is returned by the storage layer.

Reply:
Alternatively you could have a `guide: String?` field, instead, and just stick to one data class type. But yeah, let's stay consistent with two classes.

For bookmarks, for example, we do have the same split (`BookmarkNode` with guid vs `BookmarkInfo` without).